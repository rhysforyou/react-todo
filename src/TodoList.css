.todo-list {
  padding: 0;
  border-bottom: 1px solid #ddd;
}
