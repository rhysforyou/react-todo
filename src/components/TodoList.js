import PropTypes from 'prop-types';
import React from 'react'
import './TodoList.css'
import TodoItem from './TodoItem'

const renderItems = (items, onClickItem) => {
  return items.map((item, index) => (
    <TodoItem
      title={item.title}
      completed={item.completed}
      key={item.id}
      onClick={() => onClickItem(item.id)} />
  ))
}

const TodoList = ({items, onClickItem}) => (
  <ul className='todo-list'>
    {renderItems(items, onClickItem)}
  </ul>
)
TodoList.propTypes = {
  items: PropTypes.array.isRequired,
  onClickItem: PropTypes.func
}

export default TodoList