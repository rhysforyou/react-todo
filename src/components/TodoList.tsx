import React from 'react'
import './TodoList.css'
import TodoItem from './TodoItem'
import { ToDoItem as ToDoItemType } from '../reducers/todos'

interface ToDoListProps {
  items: ToDoItemType[]
  onClickItem: (itemID: string) => void
}

const renderItems = (
  items: ToDoItemType[],
  onClickItem: (itemID: string) => void
) => {
  return items.map(item => (
    <TodoItem
      title={item.title}
      completed={item.completed}
      key={item.id}
      onClick={() => onClickItem(item.id)}
    />
  ))
}

const TodoList = ({ items, onClickItem }: ToDoListProps) => (
  <ul className="todo-list">{renderItems(items, onClickItem)}</ul>
)

export default TodoList
