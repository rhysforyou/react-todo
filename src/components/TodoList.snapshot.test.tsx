/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import TodoList from './TodoList'

it('renders correctly', () => {
  let items = [
    { title: 'Item One', completed: false, id: 0 },
    { title: 'Item Two', completed: true, id: 1 },
    { title: 'Item Three', completed: true, id: 2 }
  ]
  const tree = renderer.create(<TodoList items={items} />).toJSON()
  expect(tree).toMatchSnapshot()
})
