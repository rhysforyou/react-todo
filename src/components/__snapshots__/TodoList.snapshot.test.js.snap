exports[`test renders correctly 1`] = `
<ul
  className="todo-list">
  <li
    className="todo-list-item incomplete"
    onClick={[Function]}>
    <label>
      <input
        defaultChecked={false}
        type="checkbox" />
      Item One
    </label>
  </li>
  <li
    className="todo-list-item completed"
    onClick={[Function]}>
    <label>
      <input
        defaultChecked={true}
        type="checkbox" />
      Item Two
    </label>
  </li>
  <li
    className="todo-list-item completed"
    onClick={[Function]}>
    <label>
      <input
        defaultChecked={true}
        type="checkbox" />
      Item Three
    </label>
  </li>
</ul>
`;
